tools/toolchains/arm.py

```diff
@@ -353,13 +353,17 @@ def __init__(self, target, notify=None, macros=None,
                  silent=False, extra_verbose=False, build_profile=None,
                  build_dir=None):
         target.default_toolchain = "uARM"
-        ARM.__init__(self, target, notify, macros, build_dir=build_dir,
-                     build_profile=build_profile)
-        if not set(("ARM", "uARM")).intersection(set(target.supported_toolchains)):
+
+        #At this point we already know that we want to use ARMC5+Microlib, so check for if they are supported
+        #For, AC6+Microlib we still use ARMC6 class
+        if not set(("uARM","ARMC5")).intersection(set(target.supported_toolchains)) == set(("uARM","ARMC5")):
```

Comment:
Nit: I'd recommend the following equivalent check:
```suggestion
        if not set(("uARM","ARMC5")).issubset(set(target.supported_toolchains)):
```

Reply:
Fixed